cmd/whereabouts.go

```diff
@@ -79,8 +79,12 @@ func cmdDel(args *skel.CmdArgs) error {
 
 	_, err = storage.IPManagement(types.Deallocate, *ipamConf, args.ContainerID)
 	if err != nil {
-		logging.Errorf("Error deallocating IP: %s", err)
-		return fmt.Errorf("Error deallocating IP: %s", err)
+		if (strings.Contains(fmt.Sprintf("%s", err), "Did not find reserved IP for container")) {
```

Comment:
I would opt for returning a type we can compare against instead of string comparison: https://github.com/dougbtv/whereabouts/blob/d879bd867ede305511260028ae1c4c2f049271f3/pkg/allocate/allocate.go#L61-L63


Something like:
```go
  // IPNotFoundError represents an error finding a specific IP during the deallocation process for a container.
  type IPNotFoundError struct {
      containerID string
  }

  func (e *IPNotFoundError) Error() string {
      return fmt.Sprintf("Did not find reserved IP for container %v", e.containerID)
  }
```

and to compare:

```go
    _, err = storage.IPManagement(types.Deallocate, *ipamConf, args.ContainerID)
    if err != nil {
        if notFound := &(allocate.IPNotFoundError{}); errors.As(err, &notFound) {
		logging.Debugf("Cannot deallocate IP, it might not have been assigned by us. %s", err)
	} else {
            logging.Errorf("Error deallocating IP: %s", err)
            return fmt.Errorf("Error deallocating IP: %s", err)
        }
    }
```

- https://godoc.org/errors

Reply:
@crandles Thanks!

Sounds good to me. 

Also, I realized this PR has 2 unrelated patches clubbed together:

1. 5cb45d9: To make whereabouts use hostNetwork by default
2. 7083aa4: Handle non-existent IP del request gracefully

I'll split them into two PRs for better workflow and incorporate above comment to submit again.

Thanks!